spring-integration-core/src/main/java/org/springframework/integration/store/SimpleMessageStore.java

```diff
@@ -58,6 +58,12 @@
 	private volatile boolean isUsed;
 
 	private volatile boolean copyOnGet = false;
+	
+	private final int maxWaitInMilliseconds;
```

Comment:
I'd call it like `upperBoundTimeout` and in `long` type. 
See `UpperBound.tryAcquire()`.